pkgs/development/libraries/catboost/default.nix

```
@@ -79,24 +97,45 @@ stdenv.mkDerivation (finalAttrs: {
   cmakeFlags = [
     "-DCMAKE_BINARY_DIR=$out"
     "-DCMAKE_POSITION_INDEPENDENT_CODE=on"
-    "-DCATBOOST_COMPONENTS=app;libs${lib.optionalString pythonSupport ";python-package"}"
+    "-DCATBOOST_COMPONENTS=app;libs${lib.optionalString pythonSupport ";python-package"}${lib.optionalString rLibrary ";R-package"}"
   ] ++ lib.optionals cudaSupport [
-    "-DHAVE_CUDA=on"
+    "-DHAVE_CUDA=yes"
   ];
 
+  preConfigure = lib.optionals rLibrary ''
+    export R_LIBS_SITE="$R_LIBS_SITE''${R_LIBS_SITE:+:}$out/library"
+  '';
+
+  postConfigure = lib.optionals rLibrary ''
+    # catboost's configure step deletes the contents of the R-package directory
+    # and this step replaces it from src so it can be build for rLibrary
+    cp -r $src/catboost/R-package/* catboost/R-package
+  '';
+
   installPhase = ''
     runHook preInstall
-
+  '' + lib.optionalString (!rLibrary) ''
     mkdir $dev
     cp -r catboost $dev
     install -Dm555 catboost/app/catboost -t $out/bin
```

```suggestion
    mkdir $dev
    cp -r catboost $dev
  '' + lib.optionalString (!rLibrary) ''
    install -Dm555 catboost/app/catboost -t $out/bin
```
I guess the python library is not built with `rLibrary=true; pythonSupport=true;`.

I think the installPhase will need to be reconstructed later.